scripts/generate_docs_toc.py

```diff
@@ -0,0 +1,165 @@
+
+import bigframes.bigquery
+import bigframes.geopandas
+import bigframes.pandas
+
+REPO_ROOT = pathlib.Path(__file__).parent.parent
+REFERENCE = REPO_ROOT / "docs" / "reference"
+BIGFRAMES_BIGQUERY_DIR = REFERENCE / "bigframes.bigquery"
+BIGFRAMES_BIGQUERY_TEMPLATE = """===========================
+BigQuery built-in functions
+===========================
+.. currentmodule:: bigframes.bigquery
+
+.. autosummary::
+    :toctree: generated/
+
+"""
+
+BIGFRAMES_GEOPANDAS_DIR = REFERENCE / "bigframes.geopandas"
+BIGFRAMES_GEOPANDAS_TEMPLATE = """===============================
+BigQuery DataFrames (geopandas)
+===============================
+.. currentmodule:: bigframes.geopandas
+
+.. autosummary::
+    :toctree: generated/
+
+"""
+
+BIGFRAMES_PANDAS_DIR = REFERENCE / "bigframes.pandas"
+BIGFRAMES_PANDAS_TEMPLATE = """============================
+BigQuery DataFrames (pandas)
+============================
+.. currentmodule:: bigframes.pandas
+
+.. autosummary::
+    :toctree: generated/
+
+"""
+
+
+excluded_class_attributes = frozenset(
+    {
+        "__annotations__",
+        "__class__",
+        "__delattr__",
+        "__dict__",
+        "__dir__",
+        "__doc__",
+        "__format__",
+        "__getattr__",
+        "__getattribute__",
+        "__getnewargs__",
+        "__hash__",
+        "__init__",
+        "__init_subclass__",
+        "__module__",
+        "__new__",
+        "__reduce__",
+        "__reduce_ex__",
+        "__setattr__",
+        "__slots__",
+        "__subclasshook__",
+        "__weakref__",
+    }
+)
+
+
+def find_class_attributes(cls, name):
+    # Use __dict__ instead of dir() to omit methods only implemented in a superclass.
+    # See: https://stackoverflow.com/a/7752095/101923
+    for attribute in cls.__dict__:
+        if (attribute in excluded_class_attributes) or (
+            attribute.startswith("_") and not attribute.startswith("__")
+        ):
+            continue
+
+        yield f"{name}.{attribute}"
+
+
+# Even with excluding the .nox directory in the docs/conf.py file, Sphinx still
+# complains when we alias pandas objects directly.
+excluded = frozenset(
+    {
+        "ArrowDtype",
+        "BooleanDtype",
+        "Float64Dtype",
+        "Int64Dtype",
+        "StringDtype",
+        "NA",
+    }
+)
+
+
+def create_toc_lines(module):
+    lines = []
+    attributes = list(sorted(module.__all__, key=lambda value: value.casefold()))
+
+    while len(attributes) > 0:
+        attribute = attributes.pop()
+        if attribute in excluded:
+            continue
+
+        lines.append(f"    {attribute}\n")
+
+        # Avoid trying to call getattr if this is an attribute on a class, not the
+        # module.
+        if "." in attribute:
+            continue
+
+        value = getattr(module, attribute)
+        if inspect.isclass(value):
+            for class_attribute in find_class_attributes(value, attribute):
+                attributes.insert(0, class_attribute)
```

Comment:
Tip from @arwas11 : https://docs.python.org/3/library/collections.html#collections.deque could be faster.